ISISScientificComputing/autoreduce
# Refactor queue processors

### Summary of work
- Splits Queue Processor into two separate classes `StompClient` and `MessageHandler`
- Refactor to use exceptions rather than catching the base exception (which would include say OOM exceptions)
- Fix a bug where we use run number without RB number to select a reduction record
- Consolidate some duplicated code
- More tests now things throw so are observable
- (TODO) Rename packages based on feedback

### How to test your work
- Code Review
- If possible, run a queue processor in a test environment and check everything still works for the moment

### Additional comments

Next up:
- adding a way to disconnect a Stomp Client gracefully
- Timeout period for shutting down gracefully so that cron can startup a new instance
- File Lock to ensure we only have one instance


**Release notes will be updated when work finishes**
